### DISCUSSION TITLE
actual CSS in template strings?

### DISCUSSION THREAD

**Discussion by wmertens** [johanholmerin/style9#91] [Ideas]
Seems to me that it would be possible to provide a `css` template function that converts CSS into the object format.

So e.g.

```ts
style9.create({
  blue: {
    color: 'blue',
    '@media (min-width: 80em)': {
      color: 'purple'
    }
  }
});
```

could be written as

```ts
style9.create`
  .blue {
    color: blue;
    @media (min-width: 80em) {
      color: purple;
    }
  }
`;
```

Furthermore, shorthand attributes could be used, and they would be converted to their longhand versions, with proper overriding of attributes as would be expected from object notation.

Thoughts? I already implemented [something like this for vanilla-extract](https://github.com/wmertens/styled-vanilla-extract#combined), so this is more of a "is this desirable" discussion :-)

**Comment by johanholmerin** [comment]
Using strings would mean giving up type safety, which is one of the [principles of style9](https://github.com/johanholmerin/style9/blob/master/docs/Background.md#type-safety). If you want to do so you could use something like [css-to-js.macro](https://github.com/johanholmerin/css-to-js.macro).

Using string literals would have no impact on shorthand support.
